api/src/controllers/kots/KotsAPI.ts

```diff
@@ -688,20 +689,23 @@ export async function uploadUpdate(stores, slug, buffer, source) {
   };
 }
 
-export async function syncLicense(stores, app, airgapLicense) {
+export async function syncLicense(stores, app, airgapLicense: string) {
   const license = await stores.kotsLicenseStore.getAppLicenseSpec(app.id);
 
   if (!license) {
     throw new ReplicatedError(`License not found for app with an ID of ${app.id}`);
   }
 
   let latestLicense;
-  if (airgapLicense && airgapLicense !== "") {
-    if (!app.isAirgap) {
-      throw new ReplicatedError(`Failed to sync license, app with id ${app.id} is not airgap enabled`);
+  if (app.isAirgap) {
+    if (airgapLicense === "") {
+      throw new ReplicatedError(`Failed to sync license, app with id ${app.id} is airgap enabled and no license data supplied`);
     }
     latestLicense = await verifyAirgapLicense(airgapLicense)
   } else {
+    if (airgapLicense !== "") {
```

Comment:
This function is not just for airgap, is it? if not, then we shouldn't throw an error here.

Reply:
@salahalsaleh1993 this is existing logic.  I don't want to change it.

Reply:
oh, my bad, i saw this as `===` instead of `!==`.